src/guest_memory.rs

```diff
@@ -113,6 +114,30 @@ impl_address_ops!(MemoryRegionAddress, u64);
 /// Type of the raw value stored in a GuestAddress object.
 pub type GuestUsize = <GuestAddress as AddressValue>::V;
 
+/// Describes a file mapping backing a `GuestMemoryRegion`.
+#[derive(Debug)]
+pub struct FileMappingConfig {
+    file: File,
+    offset: usize,
```

Comment:
How about using libc::off_t instead of usize here? Offset into file object may be differrent from usize.

Reply:
I would keep usize at the `guest_memory` level, in order to keep the cast into `libc::off_t` internal to the crate.

Reply:
Ultimately, like Paolo mentioned on the discussions around the original PR, `usize` turns into a bad idea here because of files larger than 4G on 32 bit systems. I'll prob end up using a type alias for offset which will be equivalent to `off_t` on Linux, and something else on Windows, just to have a clean interface.